packages/components/src/site-thumbnail/use-dominant-color.tsx

```diff
@@ -0,0 +1,42 @@
+import FastAverageColor, { FastAverageColorResult } from 'fast-average-color';
+import { useEffect, useState } from 'react';
+
+export default function useDominantColor(
+	url?: string,
+	imageRef?: React.RefObject< HTMLImageElement >
+) {
+	const [ color, setColor ] = useState< FastAverageColorResult >();
+
+	useEffect( () => {
+		const fac = new FastAverageColor();
+		//get the dominant color of a loaded image
+		if ( imageRef && imageRef.current ) {
+			const color = fac.getColor( imageRef.current, {
+				ignoredColor: [
+					[ 255, 255, 255, 255 ], // white
+					[ 0, 0, 0, 255 ], // black
+				],
+				width: 50,
+				height: 50,
+			} );
+			setColor( color );
+		}
+
+		//get the dominant color of image from url
+		if ( url && url.length > 0 ) {
+			fac
+				.getColorAsync( url, {
+					ignoredColor: [
```

Comment:
Here is my PR of my experiment: https://github.com/Automattic/wp-calypso/pull/66261/files

Reply:
> I've tried passing the image ref instead of the URL, but I'm getting the same CORS error.

The image ref option is really for local images that are not loaded from a server, I should probably remove that aspect for now. 
Even though CORS error is encountered the default background color is still used, It might be good to suppress those errors and return the default background color.


> I think accepting two optional parameters can be confusing.
> I suggest accepting only one image: string | React.RefObject< HTMLImageElement > and make a if else if based on the param type and its value.
> Reading the current code seems that two parameters can be used simultaneously.

Yeah that's true. I don't think we really need the image ref at the moment, I'm only gonna use the url for now and can extend later to accept an image reference if needs be.